src/zenml/cli/secret.py

```diff
@@ -39,472 +39,483 @@
     from zenml.secrets_managers.base_secrets_manager import BaseSecretsManager
 
 
-# Secrets
-@cli.group(cls=TagGroup, tag=CliCategories.IDENTITY_AND_SECURITY)
-@click.pass_context
-def secret(ctx: click.Context) -> None:
-    """List and manage your secrets.
-
-    Args:
-        ctx: Click context.
-    """
-    repo = Repository()
-    active_stack = repo.zen_store.get_stack(name=repo.active_stack_name)
-    secrets_manager_wrapper = active_stack.get_component_wrapper(
-        StackComponentType.SECRETS_MANAGER
-    )
-    if secrets_manager_wrapper is None:
-        error(
-            "No active secrets manager found. Please create a secrets manager "
-            "first and add it to your stack."
-        )
-        return
-
-    ctx.obj = secrets_manager_wrapper.to_component()
-
+def register_secrets_manager_subcommands() -> None:
+    """Registers CLI subcommands for the Secrets Manager."""
+    secrets_manager_group = cast(TagGroup, cli.commands.get("secrets-manager"))
+    if secrets_manager_group:
```

Comment:
tiny hint for reducing the indentation here:

```suggestion
    if not secrets_manager_group:
        return
        
    # the rest of the code follows with less indentation
```